From da111b9773a0e0ac72e0ac45fee51aaa18747a40 Mon Sep 17 00:00:00 2001
From: Kaleb Barrett <dev.ktbarrett@gmail.com>
Date: Sun, 14 Mar 2021 18:58:55 -0500
Subject: [PATCH 36/N] Fix msys library name in import libraries

Cygwin's speclib doesn't handle dashes or dots. So we specifically fix up all the import libraries that link against `msys_2_0.dll` to correctly link against `msys-2.0.dll`.
---
 winsup/cygwin/speclib | 1 +
 1 file changed, 1 insertion(+)

diff --git a/winsup/cygwin/speclib b/winsup/cygwin/speclib
index b900960..617a9c2 100755
--- a/winsup/cygwin/speclib
+++ b/winsup/cygwin/speclib
@@ -37,6 +37,7 @@ while (<$nm_fd>) {
     study;
     if (/ I _?(.*)_dll_iname/o) {
 	$dllname = $1;
+	$dllname =~ s/_2_0/-2.0/;
     } else {
 	my ($file, $member, $symbol) = m%^([^:]*):([^:]*(?=:))?.* T (.*)%o;
 	next if !defined($symbol) || $symbol =~ $exclude_regex;
-- 
2.31.0

